sympy/core/evalf.py

```diff
@@ -369,9 +368,8 @@ def calc_part(re_im, nexpr):
                     re_im, size, options)
                 assert not iim
                 nexpr = ire
-                n, c, p, b = nexpr
-                is_int = (p == 0)
-                nint = int(to_int(nexpr, rnd))
+            nint = int(to_int(nexpr, rnd))
+            is_int = (ire[2] == 0)
```

Comment:
`evalf` can return `None, None, None, None`. This happens most often when `chop=True` and the result has been chopped to 0. In this case, for `ire[2]`, we would get a `TypeError: 'NoneType' object is not subscriptable`. 

Are we sure that this case will never happen?

Reply:
I am not sure whether this case would happen or not, but I don't think it should be an issue as -
```py
Python 3.8.5 (default, Sep  4 2020, 07:30:14) 
[GCC 7.3.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> a = None
>>> a
>>> a == 0
False
```

Reply:
While I am not sure myself whether this case would happen or not, I can make a few observations - 
```py
Python 3.8.5 (default, Sep  4 2020, 07:30:14) 
[GCC 7.3.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> a = None
>>> n, c, p, b = a
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: cannot unpack non-iterable NoneType object
>>> from mpmath.libmp import to_int
>>> int(to_int(a))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/sidharth/anaconda3/envs/sympy-dev/lib/python3.8/site-packages/mpmath/libmp/libmpf.py", line 349, in to_int
    sign, man, exp, bc = s
TypeError: cannot unpack non-iterable NoneType object
```

Seeing that `evalf` returning `None` (specifically for the placeholder `ire`) would also cause these errors, which I don't think have been reported in any open issues, it should follow that this case never happens, right?

I'm not too familiar with this part of the code myself and I don't think this is the best way to prove that this case never happens, but it should be _a_ proof IMO.

Reply:
I had a bit of a look at the code, and I think it should be fine. If `ire` is None, it would have been picked up by the if statement on line 335. Changing precision between `evalf` I don't think makes a difference. Fingers crossed that everything works out.